source/extensions/filters/listener/tls_inspector/tls_inspector.cc

```diff
@@ -175,6 +175,12 @@ Network::FilterStatus Filter::onData(Network::ListenerFilterBuffer& buffer) {
   return Network::FilterStatus::StopIteration;
 }
 
+void Filter::setFilterState(FilterState::ErrorType error_type) {
+  cb_->filterState().setData(FilterState::key, std::make_unique<FilterState>(error_type),
```

Comment:
We need to check if there is already data for `ReadOnly` filter data for safety, e.g.,

```
cb_->filterState().hasData<FilterState>(FilterState::key())) {
```

If it is can be overrode, we need to use `Mutable` type.